lib/cartopy/io/ogc_clients.py

```diff
@@ -643,21 +678,41 @@ def _wmts_images(self, wmts, layer, matrix_set_name, extent,
                 # Get the tile's Image from the cache if possible.
                 img_key = (row, col)
                 img = image_cache.get(img_key)
+
                 if img is None:
-                    try:
-                        tile = wmts.gettile(
-                            layer=layer.id,
-                            tilematrixset=matrix_set_name,
-                            tilematrix=str(tile_matrix_id),
-                            row=str(row), column=str(col),
-                            **self.gettile_extra_kwargs)
-                    except owslib.util.ServiceException as exception:
-                        if ('TileOutOfRange' in exception.message and
-                                ignore_out_of_range):
-                            continue
-                        raise exception
-                    img = Image.open(io.BytesIO(tile.read()))
-                    image_cache[img_key] = img
+                    # Try it from disk cache
+                    if self.cache_path is not None:
+                        filename = "_".join([str(i) for i in img_key]) + ".npy"
```

Comment:
Might be more explicit to fill it in as an f-string since there are only two values?
```suggestion
                        filename = f"{img_key[0]}_{img_key[1]}.npy"
```